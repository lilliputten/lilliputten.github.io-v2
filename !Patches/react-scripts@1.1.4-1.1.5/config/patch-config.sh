#!/sbin/sh
# @overview Patch tool (patch-tool) config file
# @since 2018.02.25, 21:59
# @version 2018.02.25, 21:59

# Root folder
ROOTCD="node_modules"

# Target path relative to root
PATCHCD="react-scripts/config"
