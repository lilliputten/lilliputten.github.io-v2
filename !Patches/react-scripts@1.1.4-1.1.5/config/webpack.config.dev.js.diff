--- ./webpack.config.dev.js.orig	2018-09-10 19:44:07.491077400 +0300
+++ ./webpack.config.dev.js	2018-09-10 19:43:52.138444400 +0300
@@ -80,7 +80,7 @@
     // We placed these paths second because we want `node_modules` to "win"
     // if there are any conflicts. This matches Node resolution mechanism.
     // https://github.com/facebookincubator/create-react-app/issues/253
-    modules: ['node_modules', paths.appNodeModules].concat(
+    modules: [paths.appSrc, 'node_modules', paths.appNodeModules].concat(
       // It is guaranteed to exist because we tweak it in `env.js`
       process.env.NODE_PATH.split(path.delimiter).filter(Boolean)
     ),
