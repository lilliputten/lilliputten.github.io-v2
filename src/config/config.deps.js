/* eslint-disable semi */
({
  mustDeps : [
  ],
  shouldDeps : [
    { elem : 'css' },
    { elem : 'site' },
  ]
})
