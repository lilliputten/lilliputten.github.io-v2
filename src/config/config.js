/* eslint-env commonjs, browser, node */
/* globals modules */
/**
 * @module config
 * @author lilliputten <lilliputten@yandex.ru>
 * @since 2018.09.26, 23:11
 * @version 2018.11.29, 01:29
 *
 */

(function(__global){

  /** List of used submodules */
  var __submodules = [
    'css',
    'site',
    'pages',
  ]

  /** makeConfig ** {{{ Make config object from modules list
   * @param {Array} submodules
   * @return {Object}
   */
  function makeConfig (submodules) {

    var config = {}

    submodules.map(function(module, i){
      var id = __submodules[i]
      return config[id] = module
    })

    // Export to global scope
    __global.config = config

    return config
  }/*}}}*/

  // /** modules ** {{{ DEBUG: Fake modules object
  //  */
  // /* eslint-disable no-console, no-debugger */
  // var modules = {
  //   define: function (id, list, cb) {
  //     var provide = function(module) {
  //       console.log('provide', module);
  //     };
  //     console.log('define', id, list, cb);
  //     cb && cb.apply(window, [provide].concat(list));
  //   },
  // };/*}}}*/

  /** Universal export... ** {{{ */
  if (typeof module === 'object' && typeof module.exports === 'object') {

    /** submodules ** {{{ Make config modules list
     */
    var submodules = __submodules.map(function(id){

      // require commonjs module...
      try {
        var data = require('./__' + id + '/config__' + id + '.js')
      }
      catch(err) {
        var msg = 'Cannot load config module \'' + id + '\': '
        console.error(msg, err) // eslint-disable-line no-console
        debugger // eslint-disable-line no-debugger
        throw new Error(msg + (err.message || err))
      }

      // Export indiviudual __submodules...
      module.exports[id] = data

      return data

    })/*}}}*/

    // Make config object & store to global scope
    module.exports = makeConfig(submodules)
    module.exports.__submodules = __submodules

  }/*}}}*/
  /** YM export... ** {{{ */
  if (typeof modules === 'object' && typeof modules.define === 'function') {

    var configModuleNames = __submodules.map(function(id){ return 'config__' + id })

    modules.define('config', configModuleNames, function(provide /*, modules... */) {

      // Make config modules list from passed in arguments
      var submodules = Array.from(arguments).splice(1)

      // Make config object & store to global scope
      provide(makeConfig(submodules))

    })
  }/*}}}*/

})(typeof global !== 'undefined' ? global : typeof module !== 'undefined' ? module : typeof window !== 'undefined' ? window : this)
