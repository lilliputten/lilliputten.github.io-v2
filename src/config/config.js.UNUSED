/* eslint-env commonjs, browser, node */
/**
 * @module config
 * @author lilliputten <lilliputten@yandex.ru>
 * @since 2018.09.26, 23:11
 * @version 2018.11.29, 01:29
 *
 */

/** List of used submodules */
var __submodules = [
  'css',
  'site',
  'pages',
]

/** makeConfig ** {{{ Make config object from modules list
 * @param {Array} submodules
 * @return {Object}
 */
function makeConfig (submodules) {

  var config = {}

  submodules.map(function(module, i){
    var id = __submodules[i]
    return config[id] = module
  })

  // Export to global scope
  __global.config = config

  return config
}/*}}}*/

/** submodules ** {{{ Make config modules list
 */
var submodules = __submodules.map(function(id){

  // require commonjs module...
  try {
    var data = require('./__' + id + '/config__' + id + '.js')
  }
  catch(err) {
    var msg = 'Cannot load config module \'' + id + '\': '
    console.error(msg, err) // eslint-disable-line no-console
    debugger // eslint-disable-line no-debugger
    throw new Error(msg + (err.message || err))
  }

  // Export indiviudual __submodules...
  module.exports[id] = data

  return data

})/*}}}*/

// Make config object & store to global scope
module.exports = makeConfig(submodules)
module.exports.__submodules = __submodules
