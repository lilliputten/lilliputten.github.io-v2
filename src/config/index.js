/* eslint-env es6, node, commonjs */

const config = require('./config.js');
// const { css } = config;
//
// export default config;
// export { css };
// export {...config};
module.exports = config;
module.exports.site = config.site;
module.exports.css = config.css;
