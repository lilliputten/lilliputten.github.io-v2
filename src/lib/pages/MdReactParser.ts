import * as React from 'react';

import PageTools from 'lib/pages/PageTools';

import CodeBlock from 'blocks/content/CodeBlock/CodeBlock';
import Gallery from 'blocks/content/Gallery/Gallery';
import List from 'blocks/content/List/List';
import Spinner from 'blocks/content/Spinner/Spinner';

/**
 * @see https://github.com/rexxars/react-markdown
 * @see https://github.com/remarkjs/remark/blob/master/doc/plugins.md
 * @see https://github.com/zestedesavoir/zmarkdown/tree/master/packages/remark-comments
 */
// const ReactMarkdown = require('react-markdown');
const ReactMarkdown = require('react-markdown/lib/with-html');
const remarkCustomComments = require('./remark-custom-comments');
const yaml = require('js-yaml');

export type TParseResult = React.Component; // | string | null;
export interface IParseProps {
  source: string;
  className?: string;
}

export default class MdReactParser {

  private pageTools = new PageTools();

  /** plugins[] */
  private plugins: Array<any | null> = [
    [ remarkCustomComments ],
  ];

  /** customTagHandlers */
  private customTagHandlers: { [id: string]: any } = {
    COMMENTS: null,
    SPINNER: (props: any) => React.createElement(Spinner),
    GALLERY: (props: any) => React.createElement(Gallery, {...props.data, id: props.id}),
  };

  /** renderers{} ** {{{
   * @see node_modules/react-markdown/lib/renderers.js
   */
  private renderers = {
    /** customComments ** {{{
     */
    customComments: (props: any) => {
      const result = (props.tag && this.customTagHandlers[props.tag]) ? this.customTagHandlers[props.tag] : null;
      return (typeof result === 'function') ? result(props) : result;
    },
    /*}}}*/
    // /** heading ** {{{ (SAMPLE!)
    //  */
    // heading: (props: any) => {
    //   const inner = React.createElement('span', {className: 'inner'}, props.children);
    //   return React.createElement('h'.concat(props.level), {}, inner);
    // },
    // /*}}}*/
    /** text ** {{{
     */
    text: (props: any) => {
      const text = (typeof props.children === 'string') ? this.pageTools.smartypants(props.children) : props.children;
      return text;
    },
    /*}}}*/
    /** code ** {{{
     */
    code: (props: any) => {
      const codeProps = {
        content: props.value,
        language: props.language,
      };
      return React.createElement(CodeBlock, codeProps, null);
    },
    /*}}}*/
    /** list ** {{{
     */
    list: (props: any) => {
      return React.createElement(List, props, props.children);
    },
    /*}}}*/
    // listItem: ListItem,
  };
  /*}}}*/

  /** parse() ** {{{
   */
    public parse(props: IParseProps): {frontmatter: any, content: React.ReactElement<any>} {
    const {plugins, renderers} = this;
    const {frontmatter, source} = this.parseFrontmatter(props.source);
    const content = React.createElement(ReactMarkdown, {
      ...props,
      escapeHtml: false,
      source,
      plugins,
      renderers,
    });
    return {frontmatter, content};
  }/*}}}*/

  /** parseFrontmatter() ** {{{
   */
    private parseFrontmatter(source: string): {frontmatter: object, source: string} {
      let frontmatter = {};
      // Has frontmatter data?
      const match = source.match(/^\s*(---|\+\+\+)\s*\n([\s\S]*)\1\s*\n\s*\n/m);
      // Parse frontmatter if found...
      if (match) {
        const found = match[0];
        const frontmatterSrc = match[2];
        frontmatter = yaml.safeLoad(frontmatterSrc);
        if (!frontmatter || typeof frontmatter !== 'object') {
          throw new Error('Cannot parse frontmatter data for ""' + frontmatterSrc + '"');
        }
        // Strip frontmatter data...
        source = source.substr(found.length);
      }
      return { frontmatter, source };
  }/*}}}*/

}
