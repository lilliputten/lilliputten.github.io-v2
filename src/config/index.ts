/* eslint-env es6, node, commonjs */

// import config from './config.js';
import config = require('./config.js');

// // TODO: Export by `__submodules` list?
// module.exports.default = config;
// if (Array.isArray(config.__submodules)) {
//   config.__submodules.map((id: string) => module.exports[id] = config[id]);
// }
//
// // Static export (due to TS errors; TODO?)
// export default config;
// export const css = config.css;
// export const site = config.site;

// TODO: Make imports compatible with VSCode Intellisense
// export const pages = config.pages;
export const css = require('./__css/config__css');
export const site = require('./__site/config__site');
export const pages = require('./__pages/config__pages');
export default { css, site, pages };
