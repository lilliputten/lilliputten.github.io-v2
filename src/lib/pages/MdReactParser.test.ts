import * as React from 'react';

import { pages as pagesConfig } from '../../config';
import MdReactParser from './MdReactParser';

/** Do test for fail cases? */
const TO_TEST_FAIL_PROMISES = false;

let mdReactParser: MdReactParser;

describe('MdReactParser', () => {

  /*{{{*/beforeAll(() => {
    mdReactParser = new MdReactParser();
  });/*}}}*/

  /*{{{*/describe('should have basic features', () => {

    /*{{{*/it('should create instance without crashing', () => {
      expect(typeof mdReactParser).toBe('object');
      expect(mdReactParser instanceof MdReactParser).toBe(true);
    });/*}}}*/

    /*{{{*/it('should have parse method', () => {
      expect(typeof mdReactParser.parse).toBe('function');
    });/*}}}*/

  });/*}}}*/

  /*{{{*/describe('parse method', () => {

    const mdContent = `
---
param: value
number: 1
---

# title

text
`;

    let mdParsed: any;

    /*{{{*/beforeAll(() => {
      mdParsed = mdReactParser.parse({ source: mdContent });
    });/*}}}*/

    /*{{{*/it('should return object', () => {
      expect(typeof mdParsed).toBe('object');
    });/*}}}*/

    /*{{{*/describe('returned object', () => {

      /*{{{*/it('should have property frontmatter', () => {
        expect(mdParsed).toHaveProperty('frontmatter')
      });/*}}}*/

      /*{{{*/it('should have property content', () => {
        expect(mdParsed).toHaveProperty('content')
      });/*}}}*/

      /*{{{*/describe('frontmatter property', () => {

        /*{{{*/it('should be object', () => {
          expect(typeof mdParsed.frontmatter).toBe('object');
        });/*}}}*/

        /*{{{*/it('should contain parsed parameters', () => {
          expect(mdParsed.frontmatter.param).toBe('value');
          expect(typeof mdParsed.frontmatter.number).toBe('number');
        });/*}}}*/

      });/*}}}*/

      /*{{{*/describe('content property', () => {

        /*{{{*/it('should be object', () => {
          expect(typeof mdParsed.content).toBe('object');
        });/*}}}*/

        /*{{{*/it('should be valid react element', () => {
          expect(React.isValidElement(mdParsed.content)).toBe(true);
        });/*}}}*/

      });/*}}}*/

    });/*}}}*/

  });/*}}}*/

});
