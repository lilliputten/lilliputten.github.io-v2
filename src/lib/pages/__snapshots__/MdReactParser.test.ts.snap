// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MdReactParser parse method returned object content property should be rendered 1`] = `
<div
  className="markdown"
>
  <h1>
    <span
      className="inner"
    >
      title
    </span>
  </h1>
  <p>
    text
  </p>
</div>
`;

exports[`MdReactParser parse method returned object frontmatter property should be object 1`] = `
Object {
  "number": 1,
  "param": "value",
}
`;
